src/hotspot/share/gc/shared/jvmFlagConstraintsGC.cpp

```diff
@@ -285,6 +285,12 @@ JVMFlag::Error SoftMaxHeapSizeConstraintFunc(size_t value, bool verbose) {
 JVMFlag::Error HeapBaseMinAddressConstraintFunc(size_t value, bool verbose) {
   // If an overflow happened in Arguments::set_heap_size(), MaxHeapSize will have too large a value.
   // Check for this by ensuring that MaxHeapSize plus the requested min base address still fit within max_uintx.
+  if (std::numeric_limits<size_t>::max() - value < MaxHeapSize) { // overflow
```

Comment:
Sorry but I still don't see how this new check is in practice any different to the existing (albeit conditional) `(value > (max_uintx - MaxHeapSize))`?